requirements.txt

```diff
@@ -1,22 +1,15 @@
-# workarounds for pip 20.0.2's buggy dependency resolver
-# which doesn't account for cross-package constraints: https://pip.pypa.io/en/stable/user_guide/#changes-to-the-pip-dependency-resolver-in-20-3-2020
-# matplotlib pulls in the latest pyparsing, nibabel pulls in the latest packaging, and the two latests are in conflict.
-# > ERROR: packaging 21.2 has requirement pyparsing<3,>=2.0.2, but you'll have pyparsing 3.0.5 which is incompatible.
-# This forces the older pip, if the user has the older pip, to behave itself.
-# This is to specially support users on Ubuntu 20.04 LTS; when Ubuntu 22.04 LTS comes out, this can be removed.
-pyparsing<3,>=2.0.2
 csv-diff>=1.0
 loguru~=0.5
 imageio>=2.31.4
 joblib~=1.0
 matplotlib>=3.3.0
-nibabel~=3.2
+nibabel~=5.2
 onnx
 # 1.7.0>onnxruntime>=1.5.1 required `brew install libomp` on macOS.
 # So, pin to >=1.7.0 to avoid having to ask users to install libomp.
 # Avoid version 1.16.0 due to: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4225
 onnxruntime>=1.7.0,!=1.16.0
-pandas>=1.1,<1.5.0
+pandas>=1.1
```

Comment:
It looks like removing the `pandas<1.5.0` pin caused 2 tests to fail due to the usage of a deprecated call to `Dataframe.append` (more info [here](https://stackoverflow.com/a/75956237)).

Notably, `append` was removed from pandas in `v2.0`, which is a different reasoning than the previous `v1.5.0` pin (see #1194). So, at the very least, we can make the upper bound a little more lax. (Though, I imagine it would be very easy to update the `.append` calls, too.)

Reply:
- PyCharm -> Right click `DataFrame.append` method -> Go to... -> Implementation
- `pandas.core.frame` -> Right click `DataFrame.append` method -> Find Usages
- Click settings cogwheel -> Scope -> Project Files 

![image](https://github.com/ivadomed/ivadomed/assets/16181459/b00711a1-3c16-4c6f-afd9-879d74897de7)

As there are ~13 results here, I think it might be better to tackle this in its own PR, and leave the upper bound as `2.0` for now.